### DISCUSSION TITLE
line breaks after assignment instead of between parentheses / around parameters

### DISCUSSION THREAD

**Discussion by macgyver** [prettier/prettier#15196] [General]
Just wondering if anybody can explain the formatting philosophy of converting a line like 
```js
  const { section, rubric, authors, tags } = await utils.upsertCommonData(mainData);
```

into this:
```js
  const { section, rubric, authors, tags } = await utils.upsertCommonData(
    mainData
  );
```

vs. this:
```js
  const { section, rubric, authors, tags } =
    await utils.upsertCommonData(mainData);
```

or even
```js
  const { section, rubric, authors, tags }
    = await utils.upsertCommonData(mainData);
```
(though I recognize that [putting operators at the start of a line](https://github.com/prettier/prettier/issues/3806) is controversial)

it seems like keeping the code on two lines vs. three is more parsimonious in terms of vertical space, and groups the concepts better by dividing the code neatly into variable names, and assigned values. Having  `mainData` dangling by itself at the end there just doesn't read that well (to me at least).

What do other folks think? Is there a bigger concept at play here that prevents this from being the behavior? Thanks for your time and insights.  

My guess is that Prettier is trying to produce the longest line it can without exceeding the maximum line length, but in cases like this my feeling is that balancing the length of the lines it produces or trying to reduce the total number of lines produced would be a _prettier_ result.

**Comment by seiyab** [comment]
Perhaps it might be a bug.
When the right hand doesn't have `await` operator, it results as you suggested.
Assuming `await` as operator as [MDN says](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await), current Prettier looks inconsistent with `+` operator.
https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAExNrmGAS2gBpMAnAVwCNzCwyBDSmACwnLTJkYHM1MAX0wBeTIwDujQlhaEANmgB0lAA55yMAMIQAtrugARRjwAUu6VGM8AlAG4AOlCeQoGbLnxFSFGnQbiLOyc3HwCwmKMUVEAqkSKKupwmjr6RiaM5pbWjPZOLuhYOHgExFBkVLT0TEEcXJg8-EKimADU0YxxCspqGtp6BlYZWYRDtnYgJCAQqt5uyKCM5OQQEgAKSwhoyCCM8lIAnttTtIxgANZwMADKjLpwADKjcMgAZnt4J+RnlzeqZ6NeMgYFQ4FM8LpCMDQeDAfI4ABFSgQeBvD5gkAAKzQAA9rnDEcjUUh3ooMQBHIlwNYrdQ7RhoAC0UDgcAAJuzJiAQdJ5IDUhZ6fJ5Fy0ASAIIwEGEagsanJJ4stFkqasGC6eQAdVYMjgaH+YDg1y2MkIADcZAcdmA0McQGbKHAAJJQDmwa5gOizcWu64wA7w5WfECqFZ4TXfVQ7UN65Jml5TUZ9Gl8QUk9FTf6cZI7aiMahwEWZuiwTWENlsZAADgADFNyHBKYQGyneGnScGeNQyxXWMgAExTSh4AAq+e26ZVIDgugLbI5bIejCgvEofDgADEOBYpYD6SwICBBIIgA
```js
const { section, rubric, authors, tags } =
  aaaaaUtils.upsertCommonData(mainData);

const { section, rubric, authors, tags } =
  +aaaaUtils.upsertCommonData(mainData);
```

**Reply by seiyab** [reply]
I'm working on it at https://github.com/prettier/prettier/pull/15204 because it get thumbs up from a member.
(Feel free to reject it or require discussions if you don't think it is a bug.)

**Comment by fisker** [comment]
created an issue instead #15208
